### PR TITLE
[CI] allow faster downloads from the Hub in CI.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

We can dramatically speed up downloads and uploads to the Hub with `hf_transfer` on high-bandwidth machines. 

More about `hf_transfer`: https://huggingface.co/docs/huggingface_hub/v0.25.0/package_reference/environment_variables#hfhubenablehftransfer.

### CODE REVIEW

**Comment by yiyixuxu** [comment]
@ydshieh can you review this PR?

File: docker/diffusers-flax-cpu/Dockerfile [hunk 1]
```diff
@@ -43,6 +43,9 @@ RUN python3 -m pip install --no-cache-dir --upgrade pip uv==0.1.11 && \
         numpy==1.26.4 \
         scipy \
         tensorboard \
-        transformers
+        transformers \
+        hf_transfer
```
Comment (ydshieh) [inline docker/diffusers-flax-cpu/Dockerfile:47]: I think this is not the way. From the doc:

> e.g.  pip install huggingface_hub[hf_transfer]

File: docker/diffusers-flax-cpu/Dockerfile [hunk 2]
```diff
@@ -43,6 +43,9 @@ RUN python3 -m pip install --no-cache-dir --upgrade pip uv==0.1.11 && \
...
+
+ENV HF_HUB_ENABLE_HF_TRANSFER=1
```
Comment (ydshieh) [inline docker/diffusers-flax-cpu/Dockerfile:49]: I would rather set this in the workflow file that runs CI instead of set in Dockerfile files.

Comment (sayakpaul) [reply docker/diffusers-flax-cpu/Dockerfile:49]: That works for me. Let me know if you have any other comments.

Comment (ydshieh) [reply docker/diffusers-flax-cpu/Dockerfile:49]: no , that's all

File: docker/diffusers-flax-cpu/Dockerfile (hunk 1 above)
Comment (Wauplin) [reply docker/diffusers-flax-cpu/Dockerfile:47]: Both are fine yes :+1: 
Using `huggingface_hub[hf_transfer]` is less risky as it delegates the potential version issues to hfh (see [here](https://github.com/huggingface/huggingface_hub/blob/c0fd4e0f7519a4e3659c836081cc7e38c0d14b35/setup.py#L40)) but for this specific case it's more than fine to just install hf_transfer (especially in a CI)

File: docker/diffusers-flax-cpu/Dockerfile (hunk 2 above)
Comment (ydshieh) [reply docker/diffusers-flax-cpu/Dockerfile:49]: BTW, my comment is not about if it works this way or another way. It's more about the explicitness and to set  environment variables in the place where we actually use it.

**Comment by yiyixuxu** [comment]
feel free to merge whenever it is ready!

**Comment by sayakpaul** [comment]
Failing because the lastest Docker images don't have `hf_transfer`. Have triggered a fresh run [here](https://github.com/huggingface/diffusers/actions/runs/11006036265) which should solve the problem.

**Pull request merged**